update-generated.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -ex
+
+SOURCE_BRANCH=${1:-master}
+GENERATED_BRANCH=${2:-generated}
+
+git clean -f -d
+
+# checkout source branch
+git checkout "$SOURCE_BRANCH" && git submodule update
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+rsync -a ./* source
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout "$GENERATED_BRANCH" && git submodule update
+for i in *; do
+    case "$i" in
+    # Do not remove the update script or common submodule
+    update|common|source)
+        continue
+        ;;
+    *)
+        rm -rf "$i"
```

Comment:
I'm thinking... Why is needed to delete files in generated branch? - `rsync` could be used to update code by new code from `sources`.

 Without deleting files, `Add deleted files to the index as well` step isn't required.

Reply:
iirc we decided (while discussing source generation in general) that we do not want to keep archives of EOL images which is why everything gets removed first before the new generated sources get added.